src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -148,6 +167,8 @@ public synchronized void disableAuditLog()
         unregisterAsListener();
         IAuditLogger oldLogger = auditLogger;
         auditLogger = new NoOpAuditLogger(Collections.emptyMap());
+        // when we disable audit logging, we should also reset options so we return default ones (from cassandra.yml)
+        auditLogOptions = DatabaseDescriptor.getAuditLoggingOptions();
```

Comment:
I am taking my last response back. I ll return to this later.

Reply:
Thanks, I also realized that actually the log dir was configurable before and there was no reset for it so there is a change of behavior unfortunately. Thanks for looking into that

Reply:
I think that for some case (not sure if it is fql or audit, I dont remeber now), I _think_ that it was a plan to be _not able_ to change the log dir. Think about that, if somebody hacks jmx, for example, he might change the destination full query logs would be written - where he is able to read them. So that was done not settable on purpose I think.

Reply:
You are right about the log dir, I misread. It is even in the docs:
https://cassandra.apache.org/doc/latest/cassandra/new/auditlogging.html#enabling-audit-logging-with-nodetool

Reply:
But still we need to think about this reset how to make the users aware of it so there is no confusion; if we decide to really keep the reset. I don't think we reset the options for FQL? Do we?

Reply:
@ekaterinadimitrova2 FQL is returning options from cassandra.yaml when FQL is disabled (1).

Here I would expect similar behavior, when I disable audit log, it should return options from cassandra.yaml.

If we want to do this in such a way that it would "persist" these options from the last time it was enabled, that is something else. I would create new ticket for this and we would cover both cases at once instead of doing it just here and then FQL would be inconsistent with audit log behaviour.

(1) https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/fql/FullQueryLogger.java#L123-L145

Reply:
I might have forgotten something as it's been a week since I was looking into the details but if we read directly from the cassandra.yaml, why do we have to do reset when we disable? And next time when we enable it just loads the default ones on enable? I see that is what the fql tool does. What do I miss? The stop method of the fqltool does not do any reset.

Reply:
I dont know what you want me to say, honestly :) It is just done like that? We can change it but thats the scope of the other ticket how I wrote.

Reply:
One more thing related to this change, it is harder to explain here, you would need to see the code so take a deeper look please, but conceptually, I might just set "options.enabled = false" upon disabling so all other options would stay there and if you enable again, we would just take these options and I would set "options.enabled = true" etc ...

but you might (re)enable your audit logger differently next time, if  you enable and disable it, if you want to enable it again, it does not necessarilly mean that all your options from the last enablement time are OK for you this time, you might change them.

So we would have to "merge" the old options with the new options in order to rewrite these which changed, but keep these which are same.

I think this just adds additional complexity and it quite confusing for users. I prefer to "enable" and "disable" and _that's that_. I do not want to keep any "previous cached options". I just want to start with clean table next time I enable.

Reply:
I think I confused you. I understand both FQL tool and AuditLog take on enable the default cassandra.yaml parameters and whatever is provided command line arguments from NodeTool. I don't want to complicate it and change that.
My question was why we don't do:
```
public AuditLogOptions getAuditLogOptions()
    {
        if(isEnabled())
            return auditLogOptions;
        else
            // otherwise get what database is configured with from cassandra.yaml
            return  DatabaseDescriptor.getAuditLoggingOptions();
    }
```
This is how we do it for FQL log. This will make the code consistent and we don't need to do a reset in disable.

Reply:
I don't see a reason not to change the getAuditLogOptions to the way FQL does it. I think this will make it also more clear to people. Now when you look at getAuditLogOptions you will know exactly in what case what is returned. Reset in disable is not wrong(it obviously leads to the same behavior) but it is not self-explanatory.

Reply:
ok I see now, I am glad we are on the same page, yeah lets explore that idea.